uPortal-content/uPortal-content-portlet/src/main/java/org/apereo/portal/portlet/dao/jpa/PortletDefinitionImpl.java

```diff
@@ -714,18 +665,59 @@ public EntityIdentifier getEntityIdentifier() {
 
     @Override
     public PortletLifecycleState getLifecycleState() {
+
+        /*
+         * NB:  A portlet definition may have a current lifecycle
+         * state of null if it has not yet been completely created.
+         */
+        PortletLifecycleState rslt = null;
+
         final Date now = new Date();
-        if (parameters.containsKey(PortletLifecycleState.MAINTENANCE_MODE_PARAMETER_NAME)) {
-            return PortletLifecycleState.MAINTENANCE;
-        } else if (expirationDate != null && expirationDate.before(now)) {
-            return PortletLifecycleState.EXPIRED;
-        } else if (publishDate != null && publishDate.before(now)) {
-            return PortletLifecycleState.PUBLISHED;
-        } else if (approvalDate != null && approvalDate.before(now)) {
-            return PortletLifecycleState.APPROVED;
-        } else {
-            return PortletLifecycleState.CREATED;
+        for (IPortletLifecycleEntry entry : lifecycleEntries) {
```

Comment:
FWIW "Use of '_' as a lambda parameter name is not allowed." _~Idea_  ([further info](https://stackoverflow.com/questions/34521690/why-is-a-single-underscore-character-an-illegal-name-for-a-lambda-parameter))